### DISCUSSION TITLE
Batching problems of Graph

### DISCUSSION THREAD

**Discussion by borismartirosyandenovo** [pyg-team/pytorch_geometric#8036] [Q&A]
Dear community,

I have problems with pytorch_geometric. I am training a graph neural network, and I encounter with a lot of problems. Firstly my Data object consist from x, edge_attr, edge_index, target and shortest_path_attr which is the shortest path matrix retrieved from Floyd-Warshall algorithm and its dimensionality is NxN where N is the number of nodes. When I try to give it to dataloader I have problems with Batching and simply tensors do not stack because of their difference of dimensions (the main problem comes from shortest path matrix). But when I pad the matrices and dataloader problem solves, I encounter another problem: my model doesn't accept the batched tensor of X attributes (node features) because the dimensions of model and data do not match if I give a batch containing more then 1 graph info. What can I do, I would appreciate any help you can provide!

**Comment by EdisonLeeeee** [comment]
If I understand correctly, you want to customize the stacking dimensions when batching tensors in PyG data, kindly refer to [here](https://pytorch-geometric.readthedocs.io/en/latest/advanced/batching.html) for more info.
